api/resolvers/wallet.js

```diff
@@ -432,6 +432,34 @@ export default {
         },
         { settings, data }, { me, models })
     },
+
+    upsertWalletCoreLightning: async (parent, { settings, ...data }, { me, models }) => {
+      return await upsertWallet(
+        {
+          schema: CoreLightningAutowithdrawSchema,
+          walletName: 'walletCoreLightning',
+          walletType: 'CORE_LIGHTNING',
+          testConnect: async ({ rune, socket }) => {
+            const options = {
+              method: 'POST',
+              headers: {
+                'content-type': 'application/json',
+                Rune: rune
+              },
+              body: JSON.stringify({ string: rune })
+            }
+
+            return await fetch(`${socket}/v1/decode`, options).then((response) => {
+              const requiredResponse = 'method (of command) equal to \'invoice\''
+              if (requiredResponse !== response.restrictions[0].alternatives[0].summary && response.restrictions.length > 1) {
+                throw new Error('rune is not for invoice only')
+              }
+            })
+          }
+        },
+        { settings, data }, { me, models })
+    },
```

Comment:
As I mentioned in the prior review, we do not want credentials that can spend from a stacker's node to be sent to the server (even if we don't store them). 

This check should be done on the client like we do with LND and be part of validation.

Reply:
Fixed and moved it to the validate function. I'm also using the demo node url for the rune check to decouple the socket and rune from the validation

Reply:
Should we be sending runes to blockstream for all of our users? Or is this a placeholder while you work on something else?

Reply:
Originally it was a placeholder to make sure it worked so I didn't have to spin up my own core lightning node again. I had to do that because Polar was giving me grief trying to get the rest api working. But I kind of feel like it would be good to decouple the 2. I just pulled that URL from their rest api example page [here](https://docs.corelightning.org/reference/post_rpc_method_resource). But idk what do you think? Also I am working on the statistics page at the moment. I've been doing these 2 at the same time

Reply:
There's [this library](https://github.com/clams-tech/rune-decoder) but it hasn't been touched in a couple years and the [health score](https://snyk.io/advisor/npm-package/rune-decoder) looks pretty low

Reply:
> But idk what do you think? 

As a user I'd be pissed the application I'm using is sending credentials for my money to other people/companies. Wouldn't you?

> There's [this library](https://github.com/clams-tech/rune-decoder) but it hasn't been touched in a couple years and the [health score](https://snyk.io/advisor/npm-package/rune-decoder) looks pretty low

The healthscore is low for non-security reasons. It's a really simple package with a single function calling a few utility functions that easy to audit.